### PR TITLE
Mongo db

### PR DESCRIPTION
**Pull request by n3ndor**

### CODE REVIEW

File: pages/api/addItem.ts
Comment (paulpessoa) [inline pages/api/addItem.ts:1]: If you run `npm run lint` you will get some erros, just put this export way to solve it.

image

File: tsconfig.json [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
     "baseUrl":".",
+>>>>>>> e3839aae389c434edd5ed340d59b764cb99e834f
     "target": "es5",
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
     "skipLibCheck": true,
     "strict": true,
     "forceConsistentCasingInFileNames": true,
     "noEmit": true,
+<<<<<<< HEAD
```
Comment (paulpessoa) [inline tsconfig.json]: duplicated variables

File: tsconfig.json [hunk 2]
```diff
@@ -25,5 +39,6 @@
     }
   },
   "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+>>>>>>> e3839aae389c434edd5ed340d59b764cb99e834f
```
Comment (paulpessoa) [inline tsconfig.json]: remove this comment

File: tailwind.config.js
Comment (paulpessoa) [inline tailwind.config.js:1]: should be a .ts file

File: styles/globals.css
Comment (paulpessoa) [inline styles/globals.css:1]: review the folder structure

Next.js 13 use the /app directory and a diferent organization.

File: package.json
Comment (paulpessoa) [inline package.json:1]: the package.json has some comments and duplicity.
set it like the file below, delete the package-lock.json and reinstall with `npm install`

```json
{
  "name": "chingu-app",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint",
    "cypress:open": "cypress open"
  },
  "dependencies": {
    "@types/node": "20.5.0",
    "@types/react": "18.2.20",
    "@types/react-dom": "18.2.7",
    "autoprefixer": "10.4.15",
    "eslint": "8.47.0",
    "eslint-config-next": "13.4.15",
    "mongodb": "^4.8.1",
    "next": "13.4.15",
    "postcss": "^8.4.28",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "tailwindcss": "^3.3.3"
  },
  "devDependencies": {
    "@types/node": "18.7.5",
    "@types/react": "16.9.17",
    "cypress": "^12.17.3",
    "typescript": "^4.6.3"
  }
}

```

File: package-lock.json
Comment (paulpessoa) [inline package-lock.json:1]: delete this file, and then run `npm install`

File: components/Nav.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const Nav = () => {
+  return (
+    <nav className="bg-blue-500 p-4 text-white">
+      <div className="container mx-auto">
+        <ul className="flex space-x-4">
+          <li>
+            <a href="/">Home</a>
```
Comment (paulpessoa) [inline components/Nav.tsx]: We should use:

````js
import Link from "next/link";

 <Link href="/">Home</Link>
 
 ```

File: README.md [hunk 4]
```diff
@@ -41,3 +125,4 @@ You can check out [the Next.js GitHub repository](https://github.com/vercel/next
 The easiest way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new?utm_medium=default-template&filter=next.js&utm_source=create-next-app&utm_campaign=create-next-app-readme) from the creators of Next.js.
 
 Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
+>>>>>>> e3839aae389c434edd5ed340d59b764cb99e834f
```
Comment (paulpessoa) [inline README.md]: remove this line:

>>>>>>> e3839aae389c434edd5ed340d59b764cb99e834f

File: README.md [hunk 5]
```diff
@@ -1,3 +1,87 @@
+<<<<<<< HEAD
```
Comment (paulpessoa) [inline README.md]: remove this line

**Review by paulpessoa — CHANGES_REQUESTED** [review]
Good job Nandor, needs some tweaking but that's normal because of the version and approach conflict between Nextjs 12 and Nextjs13.

So we should use the `/app` directory, and instead of using `index.tsx` it will be `page.tsx`, these are some examples of files that need to be adjusted for the app to work again without errors.

@pierrelstan can you take a look?

image

**Review by pierrelstan — COMMENTED** [review]
There's an error that occurred when running 
`npm run build`

Screen Shot 2023-08-21 at 8 32 27 PM

**Comment by n3ndor** [comment]
I don't know why you get the build error, I tested before pushing:
build

**Review by paulpessoa — APPROVED** [review]
lets move on

**Pull request merged**